theatre/studio/src/panels/SequenceEditorPanel/RightOverlay/MarkerDot.tsx

```diff
@@ -0,0 +1,305 @@
+  position: absolute;
+  ${dims(HIT_ZONE_SIZE_PX)};
+  z-index: 1;
+
+  cursor: ew-resize;
+
+  ${pointerEventsAutoInNormalMode};
+
+  // "All instances of this component <Mark/> inside #pointer-root when it has the .draggingPositionInSequenceEditor class"
+  // ref: https://styled-components.com/docs/basics#pseudoelements-pseudoselectors-and-nesting
+  #pointer-root.draggingPositionInSequenceEditor:not(.draggingMarker) &,
+  #pointer-root.draggingPositionInSequenceEditor &.beingDragged {
+    pointer-events: auto;
+    cursor: var(${lockedCursorCssVarName});
+  }
+
+  #pointer-root.draggingPositionInSequenceEditor:not(.draggingMarker) & {
+    pointer-events: auto;
+    cursor: var(${lockedCursorCssVarName});
+
+    // ⸢⸤⸣⸥ thing
+    // This box extends the hitzone so the user does not
+    // accidentally leave the hitzone
+    &:hover:after {
+      position: absolute;
+      top: calc(50% - ${SNAP_CURSOR_SIZE_PX / 2}px);
+      left: calc(50% - ${SNAP_CURSOR_SIZE_PX / 2}px);
+      width: ${SNAP_CURSOR_SIZE_PX}px;
+      height: ${SNAP_CURSOR_SIZE_PX}px;
+      display: block;
+      content: ' ';
+      background: url(${SnapCursor}) no-repeat 100% 100%;
+      // This icon might also fit: GiConvergenceTarget
+    }
+  }
+
+  &.beingDragged {
+    pointer-events: none !important;
+  }
+
+  &:hover + ${MarkerVisualDot}, &.beingDragged + ${MarkerVisualDot} {
+    ${dims(MARKER_HOVER_SIZE_W_PX, MARKER_HOVER_SIZE_H_PX)}
+  }
+`
+type IMarkerDotProps = {
+  layoutP: Pointer<SequenceEditorPanelLayout>
+  markerId: SequenceMarkerId
+}
+
+const MarkerDot: React.VFC<IMarkerDotProps> = ({layoutP, markerId}) => {
+  const sheetAddress = useVal(layoutP.sheet.address)
+  const marker = useVal(
+    getStudio().atomP.historic.projects.stateByProjectId[sheetAddress.projectId]
+      .stateBySheetId[sheetAddress.sheetId].sequenceEditor.markerSet.byId[
+      markerId
+    ],
+  )
+  if (!marker) {
+    // 1/10 maybe this is normal if React tries to re-render this with
+    // out of date data. (e.g. Suspense / Transition stuff?)
+    return null
+  }
+
+  return <MarkerDotDefined marker={marker} layoutP={layoutP} />
+}
+
+export default MarkerDot
+
+type IMarkerDotDefinedProps = {
+  layoutP: Pointer<SequenceEditorPanelLayout>
+  marker: StudioHistoricStateSequenceEditorMarker
+}
+
+const MarkerDotDefined: React.VFC<IMarkerDotDefinedProps> = ({
+  layoutP,
+  marker,
+}) => {
+  const sheetAddress = useVal(layoutP.sheet.address)
+  const clippedSpaceFromUnitSpace = useVal(layoutP.clippedSpace.fromUnitSpace)
+
+  const [markRef, markNode] = useRefAndState<HTMLDivElement | null>(null)
+
+  const [contextMenu] = useMarkerContextMenu(markNode, {
+    sheetAddress,
+    markerId: marker.id,
+  })
+
+  const [isDragging] = useDragMarker(markNode, {
+    layoutP,
+    marker,
+  })
+
+  return (
+    <div
+      style={{
+        position: 'absolute',
+        zIndex: zIndexes.marker,
+        transform: `translateX(${clippedSpaceFromUnitSpace(
```

Comment:
This would still render when out of bands:

https://user-images.githubusercontent.com/593118/167209870-37bd281f-ef75-467d-a966-c597cf1bc877.mp4

To fix it, check if the marker is withing `[0, val(layoutP.clippedSpace.width)]` and if not, set its `translateX(-1000) scale(0, 0)` so it disappears but still remains in the tree and doesn't cause reflow/repaint.